### DISCUSSION TITLE
Can I run pyvista on GPU?

### DISCUSSION THREAD

**Discussion by mohammad200h** [pyvista/pyvista#5232] [Q&A] [answered]
I need to perform mesh intersection on GPU to speed up the process. is that possible?

**Comment by tkoyama010** [comment, marked as answer]
PyVista uses [vtk](https://pypi.org/project/vtk/) as its engine. In vtk, GPU is used for rendering, but many parts of the pipeline run on the CPU. So I assume that the [boolean_difference](https://docs.pyvista.org/version/stable/api/core/_autosummary/pyvista.PolyDataFilters.boolean_difference.html) (using vtk's [vtkBoreanOperationPolyDataFilter](https://vtk.org/doc/nightly/html/classvtkBooleanOperationPolyDataFilter.html#details)), which you want to speed up, is also using CPU only.
